### PR TITLE
A sizeable plugin update

### LINKED ISSUE #21: VineflowerInvoker.decompile holds ReadWrite lock for 65 seconds and freezes UI thread
```
com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1100) [56000ms]
com.intellij.openapi.application.rw.CancellableReadActionKt$$Lambda$1605/0x000000080113a940.run(Unknown Source) [56000ms]
com.intellij.openapi.application.rw.CancellableReadActionKt.cancellableReadActionInternal$lambda$1$lambda$0(cancellableReadAction.kt:49) [56000ms]
com.intellij.openapi.progress.CancellationKt.executeWithJobAndCompleteIt(cancellation.kt:126) [56000ms]
com.intellij.openapi.progress.CancellationKt.withCurrentJob(cancellation.kt:17) [56000ms]
com.intellij.openapi.progress.Cancellation.withCurrentJob(Cancellation.java:60) [56000ms]
com.intellij.openapi.progress.CancellationKt$$Lambda$1594/0x0000000801131cb8.compute(Unknown Source) [56000ms]
com.intellij.openapi.progress.CancellationKt.withCurrentJob$lambda$0(cancellation.kt:17) [56000ms]
com.intellij.openapi.application.rw.InternalReadAction$tryReadCancellable$1.invoke(InternalReadAction.kt:91) [56000ms]
com.intellij.openapi.application.rw.InternalReadAction$tryReadCancellable$1.invoke(InternalReadAction.kt:92) [56000ms]
com.intellij.openapi.application.rw.InternalReadAction.access$insideReadAction(InternalReadAction.kt:15) [56000ms]
com.intellij.openapi.application.rw.InternalReadAction.insideReadAction(InternalReadAction.kt:108) [56000ms]
com.intellij.openapi.fileEditor.impl.UiBuilder$processFiles$2$document$1.invoke(EditorsSplitters.kt:969) [56000ms]
com.intellij.openapi.fileEditor.impl.UiBuilder$processFiles$2$document$1.invoke(EditorsSplitters.kt:970) [56000ms]
com.intellij.openapi.fileEditor.impl.FileDocumentManagerBase.getDocument(FileDocumentManagerBase.java:53) [56000ms]
com.intellij.openapi.fileEditor.impl.FileDocumentManagerBase.loadText(FileDocumentManagerBase.java:95) [56000ms]
com.intellij.openapi.fileEditor.impl.LoadTextUtil.loadText(LoadTextUtil.java:478) [56000ms]
com.intellij.psi.impl.compiled.ClassFileDecompiler.decompile(ClassFileDecompiler.java:29) [56000ms]
org.vineflower.ijplugin.VineflowerDecompilerLight.getText(VineflowerDecompilerLight.kt:6) [56000ms]
org.vineflower.ijplugin.VineflowerDecompilerLight.getText(VineflowerDecompilerLight.kt:12) [56000ms]
org.vineflower.ijplugin.VineflowerDecompilerBase$Companion.getText(VineflowerDecompilerBase.kt:39) [56000ms]
org.vineflower.ijplugin.VineflowerInvoker.decompile(VineflowerInvoker.kt:168) [56000ms]
org.vineflower.ijplugin.VineflowerInvoker.decompile$doDecompile(VineflowerInvoker.kt:125) [56000ms]
java.base@17.0.7/java.lang.reflect.Method.invoke(Unknown Source) [56000ms]
java.base@17.0.7/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) [56000ms]
java.base@17.0.7/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) [56000ms]
java.base@17.0.7/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) [56000ms]
org.jetbrains.java.decompiler.main.decompiler.BaseDecompiler.decompileContext(BaseDecompiler.java:44) [56000ms]
org.jetbrains.java.decompiler.main.Fernflower.decompileContext(Fernflower.java:149) [56000ms]
org.jetbrains.java.decompiler.struct.StructContext.saveContext(StructContext.java:178) [56000ms]
org.jetbrains.java.decompiler.struct.ContextUnit.save(ContextUnit.java:173) [56000ms]
org.jetbrains.java.decompiler.struct.ContextUnit.waitForAll(ContextUnit.java:226) [56000ms]
java.base@17.0.7/java.util.concurrent.FutureTask.get(Unknown Source) [56000ms]
java.base@17.0.7/java.util.concurrent.FutureTask.awaitDone(Unknown Source) [56000ms]
java.base@17.0.7/java.util.concurrent.locks.LockSupport.park(Unknown Source) [56000ms]
java.base@17.0.7/jdk.internal.misc.Unsafe.park(Native Method) [56000ms]
```

### PR DESCRIPTION
**Pull request by sschr15**
After a handful of messing around, I've succeeded in some key goals, and I think it's enough to improve the plugin enough

- Update a bunch of things
  - Gradle 7.6 -> 8.9
  - Java 11 -> 17
  - Kotlin 1.8.0 -> 1.9.22
  - IntelliJ 2023.1 -> 2024.1 (Kotlin plugin reasons)
  - Gradle IntelliJ Plugin 1.13.2 -> IntelliJ Platform Plugin 2.0.1
- Create a Full decompiler for Kotlin
  - The decompiler is a bit buggy as of now and has unexpected psi issues
- Update language checking to match that of latest Vineflower versions
- Presumably fix #21
- Change version to Semver equivalent (remains at 1.1.0)

### CODE REVIEW

**Comment by Earthcomputer** [comment]
Can you explain what was causing the race condition so I can understand your changes?

**Comment by sschr15** [comment]
I'm not fully confident on what caused the race condition exactly, but my best guess is that the read action to load a classfile was waiting on the main thread, itself waiting on the read action before passing control back to the decompiler. By using the queue system already present for cases in which read access was allowed, it seems to fix the deadlock.

The change to always using the queue was a hopeful guess that seems to work from my testing.

File: src/main/kotlin/org/vineflower/ijplugin/VineflowerInvoker.kt [hunk 1]
```diff
@@ -60,7 +59,7 @@ class VineflowerInvoker(classLoader: ClassLoader) {
         )
 
 
-    private val baseDecompilerClass = classLoader.loadClass("org.jetbrains.java.decompiler.main.decompiler.BaseDecompiler")
+    private val baseDecompilerClass = classLoader.loadClass("org.jetbrains.java.decompiler.main.Fernflower")
```
Comment (Earthcomputer) [inline src/main/kotlin/org/vineflower/ijplugin/VineflowerInvoker.kt:62]: Is this a change in VF's API? Will this work on older version of VF? We need to stay supporting older VF/QF versions. That's what the `Feature`s down below are for.

Also, these variable names need to be changed.

File: src/main/resources/META-INF/plugin.xml [hunk 2]
```diff
@@ -20,5 +21,7 @@
                 displayName="Vineflower"/>
         <applicationService serviceImplementation="org.vineflower.ijplugin.VineflowerState"/>
         <psi.classFileDecompiler implementation="org.vineflower.ijplugin.VineflowerDecompilerLight" order="first"/>
+<!--        <psi.classFileDecompiler implementation="org.vineflower.ijplugin.VineflowerDecompilerJava" order="first"/>-->
```
Comment (Earthcomputer) [inline src/main/resources/META-INF/plugin.xml:24]: This comment can be removed

**Review by Earthcomputer — CHANGES_REQUESTED** [review]

File: src/main/kotlin/org/vineflower/ijplugin/VineflowerInvoker.kt (hunk 1 above)
Comment (sschr15) [reply src/main/kotlin/org/vineflower/ijplugin/VineflowerInvoker.kt:62]: Apparently, I didn't look closely enough when trying to work around some NPEs done during debugging and updating

**Review by Earthcomputer — APPROVED** [review]

**Pull request merged**